subprojects/core/src/integTest/groovy/org/gradle/api/tasks/NestedInputIntegrationTest.groovy

```diff
@@ -778,32 +777,34 @@ class NestedInputIntegrationTest extends AbstractIntegrationSpec implements Dire
         """
     }
 
-    def "task with nested bean loaded with custom classloader is not cached"() {
-        file("input.txt").text = "data"
-        buildFile << taskWithNestedBeanFromCustomClassLoader()
-
-        when:
-        withBuildCache().run "customTask", "--info", "-D${BuildCacheDebugLoggingOption.GRADLE_PROPERTY}=true"
-        then:
-        output.contains "Caching disabled for task ':customTask' because:\n" +
-            "  Non-cacheable inputs: property 'bean' was loaded with an unknown classloader (class 'NestedBean')."
-    }
-
+    @ValidationTestFor(
+        ValidationProblemId.UNKNOWN_IMPLEMENTATION
+    )
     @ToBeFixedForConfigurationCache(because = "uses custom GroovyClassLoader")
-    def "task with nested bean loaded with custom classloader is never up-to-date"() {
+    def "task with nested bean loaded with custom classloader disables execution optimizations"() {
+        expectUnindentedValidationMessage()
```

Comment:
Let's move this to `expectThatExecutionOptimizationDisabledWarningIsDisplayed()`.

Reply:
This call is not necessary any more, I removed the method.